src/test/java/org/opensearch/neuralsearch/processor/TextEmbeddingProcessorIT.java

```diff
@@ -0,0 +1,161 @@
+            "POST",
+            String.format("/_plugins/_ml/models/%s/_load", modelId),
+            null,
+            TestHelper.toHttpEntity(""),
+            ImmutableList.of(new BasicHeader(HttpHeaders.USER_AGENT, "Kibana"))
+        );
+        JsonNode uploadResJson = objectMapper.readTree(EntityUtils.toString(uploadResponse.getEntity()));
+        String taskId = uploadResJson.get("task_id").asText();
+        assertNotNull(taskId);
+
+        JsonNode taskQueryResult = getTaskQueryResponse(taskId);
+        boolean isComplete = checkComplete(taskQueryResult);
+        while (!isComplete) {
+            taskQueryResult = getTaskQueryResponse(taskId);
+            isComplete = checkComplete(taskQueryResult);
+        }
+    }
+
+    private void createPipelineProcessor(String modelId) throws IOException {
+        Response pipelineCreateResponse = TestHelper.makeRequest(
+            client(),
+            "PUT",
+            "/_ingest/pipeline/" + pipelineName,
+            null,
+            TestHelper.toHttpEntity(
+                String.format(
+                    FileUtils.readFileToString(
+                        new File(classLoader.getResource("processor/PipelineConfiguration.json").getFile()),
+                        "utf-8"
+                    ),
+                    modelId
+                )
+            ),
+            ImmutableList.of(new BasicHeader(HttpHeaders.USER_AGENT, "Kibana"))
+        );
+        JsonNode node = objectMapper.readTree(EntityUtils.toString(pipelineCreateResponse.getEntity()));
+        assertTrue(node.get("acknowledged").asBoolean());
+    }
+
+    private void createIndex() throws IOException {
+        Response response = TestHelper.makeRequest(
+            client(),
+            "PUT",
+            indexName,
+            null,
+            TestHelper.toHttpEntity(
+                String.format(
+                    FileUtils.readFileToString(new File(classLoader.getResource("processor/IndexConfiguration.json").getFile()), "utf-8"),
+                    pipelineName
+                )
+            ),
+            ImmutableList.of(new BasicHeader(HttpHeaders.USER_AGENT, "Kibana"))
+        );
+        JsonNode node = objectMapper.readTree(EntityUtils.toString(response.getEntity()));
+        assertTrue(node.get("acknowledged").asBoolean());
+        assertEquals(indexName, node.get("index").asText());
+    }
+
+    private void ingestDocument() throws IOException {
+        Response response = TestHelper.makeRequest(
+            client(),
+            "POST",
+            indexName + "/_doc",
+            null,
+            TestHelper.toHttpEntity(
+                FileUtils.readFileToString(new File(classLoader.getResource("processor/IngestDocument.json").getFile()), "utf-8")
+            ),
+            ImmutableList.of(new BasicHeader(HttpHeaders.USER_AGENT, "Kibana"))
+        );
+        JsonNode node = objectMapper.readTree(EntityUtils.toString(response.getEntity()));
```

Comment:
check other Integration tests on how to read the responses.

Reply:
Changed to XContent approach.